### PR TITLE
[BUILD] update release scripts

### PR DESCRIPTION
**Pull request by felixcheung**
## What changes were proposed in this pull request?

Change to dist.apache.org instead of home directory
sha512 should have .sha512 extension. From ASF release signing doc: "The checksum SHOULD be generated using SHA-512. A .sha file SHOULD contain a SHA-1 checksum, for historical reasons."

NOTE: I *think* should require some changes to work with Jenkins' release build

## How was this patch tested?

manually

### CODE REVIEW

**Comment by felixcheung** [comment]
Jenkins, retest this please

File: dev/create-release/release-build.sh [hunk 1]
```diff
@@ -135,50 +143,24 @@ if [ -z "$SPARK_PACKAGE_VERSION" ]; then
...
+  echo "Packaging release source tarballs"
   cp -r spark spark-$SPARK_VERSION
   tar cvzf spark-$SPARK_VERSION.tgz spark-$SPARK_VERSION
   echo $GPG_PASSPHRASE | $GPG --passphrase-fd 0 --armour --output spark-$SPARK_VERSION.tgz.asc \
     --detach-sig spark-$SPARK_VERSION.tgz
   echo $GPG_PASSPHRASE | $GPG --passphrase-fd 0 --print-md MD5 spark-$SPARK_VERSION.tgz > \
     spark-$SPARK_VERSION.tgz.md5
   echo $GPG_PASSPHRASE | $GPG --passphrase-fd 0 --print-md \
-    SHA512 spark-$SPARK_VERSION.tgz > spark-$SPARK_VERSION.tgz.sha
+    SHA512 spark-$SPARK_VERSION.tgz > spark-$SPARK_VERSION.tgz.sha512
```
Comment (srowen) [inline dev/create-release/release-build.sh]: +1 for making it explicit as sha will often default to SHA1

File: dev/create-release/release-build.sh [hunk 2]
```diff
@@ -392,6 +375,7 @@ if [[ "$1" == "publish-release" ]]; then
   find . -type f |grep -v \.jar |grep -v \.pom | xargs rm
 
   echo "Creating hash and signature files"
+  # this must have .asc, .md5 and .sha1 - it really doesn't like anything else there
```
Comment (srowen) [inline dev/create-release/release-build.sh]: sha1, not sha512?

Comment (felixcheung) [reply dev/create-release/release-build.sh]: it rejects it when there are "extra files" of the sha512 extension.

**Comment by srowen** [comment]
OK to commit this? I'm also not clear if it will require a Jenkins change, but this change seems like a right way forward

**Comment by felixcheung** [comment]
I think we have daily Jenkins release build runs (release, doc, artifact snapshot) that have been pushing to home.apache.org

**Comment by srowen** [comment]
Are they the jobs in https://amplab.cs.berkeley.edu/jenkins/view/Spark%20Packaging/ ? I might be able to edit those if you can guide me about what needs to change. CC @JoshRosen 
Seems like we can go ahead with this in any event.

**Comment by srowen** [comment]
merged to master

**Comment by felixcheung** [comment]
yes, specifically this changes the output location for
https://amplab.cs.berkeley.edu/jenkins/view/Spark%20Packaging/job/spark-master-docs/
https://amplab.cs.berkeley.edu/jenkins/view/Spark%20Packaging/job/spark-master-package/

and later
https://amplab.cs.berkeley.edu/jenkins/view/Spark%20Packaging/job/spark-release-docs/
https://amplab.cs.berkeley.edu/jenkins/view/Spark%20Packaging/job/spark-release-package/

and it shows up like this
https://dist.apache.org/repos/dist/dev/spark/2.3.0-SNAPSHOT-2017_12_09_08_01-ab1b6ee-docs/

**Comment by srowen** [comment]
The package build was failing for a while, but the docs build has a probably-related new error:

```
ATTENTION!  Your password for authentication realm:

   <https://dist.apache.org:443> ASF Committers

can only be stored to disk unencrypted!  You are advised to configure
your system so that Subversion can store passwords encrypted, if
possible.  See the documentation for details.

You can avoid future appearances of this warning by setting the value
of the 'store-plaintext-passwords' option to either 'yes' or 'no' in
'/home/jenkins/.subversion/servers'.
-----------------------------------------------------------------------
Store password unencrypted (yes/no)? svn: Commit failed (details follow):
svn: Can't read stdin: End of file found
Deleting credential directory /home/jenkins/workspace/spark-master-docs/spark-utils/new-release-scripts/jenkins/jenkins-credentials-wmz3q7pN
Build step 'Execute shell' marked build as failure
[WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
Finished: FAILURE
```

Was this expected and needs a Jenkins config change, or not expected?

**Comment by srowen** [comment]
(The package build failure, which began Nov 26, is:)

```
gpg: skipped "/home/jenkins/workspace/spark-master-package/spark-utils/new-release-scripts/jenkins/jenkins-credentials-FLng1qGV/gpg.tmp": No secret key
gpg: signing failed: No secret key
Deleting credential directory /home/jenkins/workspace/spark-master-package/spark-utils/new-release-scripts/jenkins/jenkins-credentials-FLng1qGV
Build step 'Execute shell' marked build as failure
[WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
Finished: FAILURE
```

**Comment by felixcheung** [comment]
I think these are Jenkins specific config related to how the user credentials are stored (not) securely?

Specifically I don’t know how we are saving ASF user and password inside Jenkins to publish to svn before.

Not sure about the gpg one since I don’t think we merged any changes to scripts then. Perhaps some changes to Jenkins with regards to where the private key is stored.

Generally both sound like something we should review more closely since they are security practices.

**Pull request closed without merging**